pkg/providers/tinkerbell/template.go

```diff
@@ -47,6 +48,14 @@ func NewTinkerbellTemplateBuilder(datacenterSpec *v1alpha1.TinkerbellDatacenterC
 
 func (tb *TinkerbellTemplateBuilder) GenerateCAPISpecControlPlane(clusterSpec *cluster.Spec, buildOptions ...providers.BuildMapOption) (content []byte, err error) {
 	cpTemplateConfig := clusterSpec.TinkerbellTemplateConfigs[tb.controlPlaneMachineSpec.TemplateRef.Name]
```

Comment:
When would we ever expect this to not be nil?

Reply:
A user can define their own `TinkerbellTemplate` if they want to override the default one. I agree, we can have an optional action overrides compared to a complete custom resource but it needs some discussion and we can do that in the separate PR.